presto-main/src/main/java/com/facebook/presto/sql/gen/JoinCompiler.java

```diff
@@ -222,7 +268,13 @@ private static FieldDefinition generateInstanceSize(ClassDefinition definition)
         generateRowEqualsRowMethod(classDefinition, callSiteBinder, joinChannelTypes);
         generatePositionEqualsRowMethod(classDefinition, callSiteBinder, joinChannelTypes, joinChannelFields, true);
         generatePositionEqualsRowMethod(classDefinition, callSiteBinder, joinChannelTypes, joinChannelFields, false);
-        generatePositionEqualsRowWithPageMethod(classDefinition, callSiteBinder, joinChannelTypes, joinChannelFields);
+        if (groupByUsesEqualTo) {
+            generatePositionEqualsRowWithPageMethod(classDefinition, callSiteBinder, joinChannelTypes, joinChannelFields, "positionNotDistinctFromRow");
```

Comment:
Passing in method name is pretty confusing. I don't like the way it breaks abstraction.

What about generating a delegating method here instead?